pytorch_forecasting/models/base/_base_model_v2.py

```diff
@@ -294,3 +294,175 @@ def log_metrics(
+
+        expected_dims : tuple[int, Optional[int], Optional[int], Optional[int]], default=(None, None, None, None)
+        A tuple specifying the dimensions: (n_targets, batch_size, timesteps, last_dim)
+
+            n_targets : int
+                - Position 0: Number of target features
+                - Must be provided explicitly (cannot be None)
+                - Used for reshaping 2D and 3D tensors to 4D.
+
+            batch_size : Optional[int], default=None
+                - Position 1: Expected batch size
+                - When specified: Validates prediction.shape[0]
+                - When None: Uses actual tensor dimension
+
+            timesteps : Optional[int], default=None
+                - Position 2: Expected number of timesteps
+                - When specified: Validates prediction.shape[1]
+                - When None: Uses actual tensor dimension
+
+            last_dim : Optional[int], default=None
+                - Position 3: Size of the last dimension.
+                - Common use case - quantile, sample, distribution params.
+                - When it is specified, it is used to directly reshape.
+                - When None and model uses QuantileLoss: It is set to the number of quantiles
+                - When None and no quantile information is available: It defaults to 1.
+                - If required, this can be extended to handle other cases where the last_dim is None
+                but its value can be inferred from the loss function or model configuration (apart from
+                the existing QuantileLoss case, of course).
+        Returns
+        -------
+        torch.Tensor
+            The standardized prediction tensor with shape (batch_size, timesteps, n_targets, last_dim).
+            The prediction tensor is obtained by reshaping the input tensor. There are
+            several cases to consider:
+
+            - If the input tensor is 2D, it is reshaped to (batch_size, timesteps, 1, 1).
+            - If the input tensor is 3D, it is reshaped to (batch_size, timesteps, 1, 1) for a
+            multivariate single-target forecast, or (batch_size, timesteps, 1, last_dim) for a univariate quantile forecast.
+            - If the input tensor is 4D, it is assumed to be in the shape
+            (batch_size, timesteps, n_targets, last_dim) or (batch_size, timesteps, last_dim, n_targets).
+        Notes
+        -----
+        [1] The fourth dimension (last_dim) commonly represents:
+
+            * Quantiles: For quantile regression (e.g., 0.1, 0.5, 0.9)
+            * Distribution parameters: For parametric forecasts (e.g., mean, variance)
+            * Samples: For sample-based uncertainty estimates
+
+            The current implementation assumes the most common case of quantile forecasts
+            when automatically inferring this dimension from the loss function,
+            but any value can be explicitly provided. A fallback of 1 is used in case where
```

Comment:
not a "fallback", in this case it is simply the predictive mean (a single float)